extendr-api/src/functions.rs

```diff
@@ -385,8 +385,14 @@ pub fn na_str() -> &'static str {
 ///   Ok(())
 /// })
 /// ```
+#[cfg(feature = "extendr_engine")]
 pub fn test<F: FnOnce() -> Result<()>>(f: F) {
     extendr_engine::start_r();
 
     f().unwrap();
 }
+
+#[cfg(not(feature = "extendr_engine"))]
+pub fn test<F: FnOnce() -> Result<()>>(_: F) {
+    ()
+}
```

Comment:
Can you leave this out if all tests are properly enclosed into `#[cfg(test)]`? This would then be another check against accidentally including tests outside a tests module.